packages/language-server/src/completion/completions.ts

```diff
@@ -336,28 +352,29 @@ export function getSuggestionForFirstInsideBlock(
   }
 }
 
+/**
+ * Returns the currently available _blocks_ for completion.
+ * Currently available: Generator, Datasource, Model, Enum, View
+ * @param lines
+ * @returns the list of block suggestions
+ */
 export function getSuggestionForBlockTypes(lines: string[]): CompletionList {
   // create deep copy
-  const suggestions: CompletionItem[] = klona(allowedBlockTypes)
-
-  // enum is not supported in sqlite
-  let foundDataSourceBlock = false
-  for (const item of lines) {
-    if (item.includes('datasource')) {
-      foundDataSourceBlock = true
-      continue
-    }
-    if (foundDataSourceBlock) {
-      if (item.includes('}')) {
-        break
-      }
-      if (item.startsWith('provider') && item.includes('sqlite')) {
-        suggestions.pop()
-      }
-    }
-    if (!suggestions.map((sugg) => sugg.label).includes('enum')) {
-      break
-    }
+  let suggestions: CompletionItem[] = klona(allowedBlockTypes)
+
+  const datasourceProvider = getFirstDatasourceProvider(lines)
+  const previewFeatures = getAllPreviewFeaturesFromGenerators(lines)
+
+  const isEnumAvailable = Boolean(datasourceProvider && !datasourceProvider.includes('sqlite'))
```

Comment:
This is currently wrong - it should only _not_ be available for SQLite.

Reply:
Updated!